tests/tests_api/test_group.py

```diff
@@ -77,24 +101,180 @@ def test_public_group_retrieve(api_client):
+        response = api_client.post(self._make_url(group))
+
+        assert response.status_code == status.HTTP_201_CREATED
+
+        group.refresh_from_db()
+        assert group.status == GroupStatus.WORKING
+
+        common_status_since = group.status_since
+        self._compare_date_time_with_timestamp(common_status_since, timestamp)
+
+        for coordinator in group.coordinators.iterator():
+            assert coordinator.status in (
+                CoordinatorStatus.WORKING_BELOW_THRESHOLD,
+                CoordinatorStatus.WORKING_OK,
+                CoordinatorStatus.WORKING_LIMIT_REACHED,
+            )
+            assert coordinator.status_since == common_status_since
+
+            log_event: CoordinatorLogEvent = CoordinatorLogEvent.objects.get(
+                coordinator_id=coordinator.pk
+            )
+            assert log_event.type == CoordinatorLogEventType.TOOK_NEW_GROUP
+            self._compare_date_time_with_timestamp(log_event.date_time, timestamp)
+
+        for student in group.students.iterator():
+            assert student.status == StudentStatus.STUDYING
+            assert student.status_since == common_status_since
+
+            log_event: StudentLogEvent = StudentLogEvent.objects.get(student_id=student.pk)
+            assert log_event.type == StudentLogEventType.STUDY_START
+            self._compare_date_time_with_timestamp(log_event.date_time, timestamp)
+
+        for teacher in group.teachers.iterator():
+            assert teacher.status in (
+                TeacherStatus.TEACHING_ACCEPTING_MORE,
+                TeacherStatus.TEACHING_NOT_ACCEPTING_MORE,
+            )
+            assert teacher.status_since == common_status_since
+
+            log_event: TeacherLogEvent = TeacherLogEvent.objects.get(teacher_id=teacher.pk)
+            assert log_event.type == TeacherLogEventType.STUDY_START
+            self._compare_date_time_with_timestamp(log_event.date_time, timestamp)
+
+    @pytest.mark.parametrize(
+        "number_of_groups_to_start, expected_status",
+        [
+            (CoordinatorGroupLimit.MIN - 1, CoordinatorStatus.WORKING_BELOW_THRESHOLD),
+            (CoordinatorGroupLimit.MIN, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MIN + 1, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MAX - 1, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MAX, CoordinatorStatus.WORKING_LIMIT_REACHED),
+            (CoordinatorGroupLimit.MAX + 1, CoordinatorStatus.WORKING_LIMIT_REACHED),
+        ],
+    )
+    def test_public_group_start_coordinator_status(
+        self, api_client, timestamp, number_of_groups_to_start, expected_status
+    ):
+        coordinator = baker.make(Coordinator, _fill_optional=True)
+        coordinator.status = CoordinatorStatus.WORKING_BELOW_THRESHOLD
+        coordinator.save()
+
+        for i in range(number_of_groups_to_start):
```

Comment:
It's better to use `_` for unused variables

Reply:
Done